crates/lakekeeper/src/api/management/v1/tasks.rs

```diff
@@ -77,13 +81,170 @@ pub struct Task {
     pub updated_at: Option<chrono::DateTime<chrono::Utc>>,
 }
 
+impl TryFrom<TaskInfo> for TaskInfoAPI {
+    type Error = ErrorModel;
+
+    fn try_from(value: TaskInfo) -> Result<Self, Self::Error> {
+        let status = value
+            .status()
+            .copied()
+            .map(TaskStatus::from)
+            .or(value.outcome().copied().map(TaskStatus::from))
+            .ok_or(ErrorModel::internal(
+                "Task must have either status or outcome",
+                "StatusOrOutcomeMissing",
+                None,
+            ))?;
```

Comment:
We should make this typesafe. It is the responsibility of the DB to return either one or the other. Currently this information is implicit "in the docstring", which leaves room for human error to return neither or both. Because of the current data structure, this case would be detected at runtime and exposed to the user as an internal error. Instead, lets make it typesafe so that this error is detected at compile time.